minpy/context.py

```diff
@@ -0,0 +1,139 @@
+# coding: utf-8
+"""Context management API of minpy."""
+from __future__ import absolute_import
+
+from .utils import log
+
+_logger = log.get_logger(__name__)
+
+
+class Context(object):
+    """Constructing a context.
+
+    Parameters
+    ----------
+    device_type : {'cpu', 'gpu'} or Context.
+        String representing the device type
+
+    device_id : int (default=0)
+        The device id of the device, needed for GPU
+
+    Note
+    ----
+    Context can also be used a way to change default context.
+
+    Examples
+    --------
+    >>> import minpy as np
+    >>> # array on cpu
+    >>> cpu_array = np.ones((2, 3))
+    >>> # switch default context to GPU(2)
+    >>> with np.Context(np.gpu(2)):
+    >>>     gpu_array = np.ones((2, 3))
+    >>> gpu_array.context
+    gpu(2)
+    """
+    # static class variable
+    default_ctx = None
+    devtype2str = {1: 'cpu', 2: 'gpu', 3: 'cpu_pinned'}
+    devstr2type = {'cpu': 1, 'gpu': 2, 'cpu_pinned': 3}
```

Comment:
Could you generate `devstr2type` in case we change `devtype2str` but forgot to change the other?
```python
devstr2type = {v: k for k, v in devtype2str.items()}
```